controllers/bot.php

```
@@ -18,15 +19,18 @@ public function update( $boturl = '' ) {
                 $bot->sendInitiateRequest(); 
             }
             catch ( GraderBotException $e ) {
-                $error = end( $bot->errors );
-                $expected = $error[ 'expected' ];
-                $actual = $error[ 'actual' ];
-                $error = $error[ 'error' ];
-                $error = str_replace( "initiate_", "", $error );
-                if ( strpos( $error, '_not_set' ) ) {
-                    $error = 'invalid_json_dictionary';
+                $error = new Error();
+                $error->user = $bot->user;
+                $description = end( $bot->errors );
+                $error->expected = $description[ 'expected' ];
+                $error->actual = $description[ 'actual' ];
+                $error->error = $description[ 'error' ];
+                $error->error = str_replace( "initiate_", "", $error->error );
+                if ( strpos( $error->error, '_not_set' ) ) {
+                    $error->error = 'invalid_json_dictionary';
                 }
-                go( 'bot', 'update', [ 'bot_fail' => true, 'error' => $error, 'actual' => $actual, 'expected' => $expected ] );
+                $error->save();
```

Let it throw after it saves the error